### PR TITLE
Adding Neo.Runtime.GetTime and Neo.Runtime.GetCurrentBlock

### PR DESCRIPTION
**Pull request by localhuman**
for issue #114 

- adds `Neo.Runtime.GetTime`
- adds `Neo.Runtime.GetCurrentBlock`
- adds property `CurrentBlock` to `Blockchain`.  This retrieves the currently persisting block if a block is currently being persisted, otherwise it returns the highest block

If either of these API are called during `TriggerType.Verification`, the most recent blockchain on the block is returned.  If it is called during `TriggerType.Application`, it will be the currently persisting block.

### CODE REVIEW

File: neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.cs [hunk 1]
```diff
@@ -106,6 +108,18 @@ public LevelDBBlockchain(string path)
...
 
+        public override Block CurrentBlock
+        {
+            get
+            {
+                if (persisting_block != null)
+                {
+                    return persisting_block;
+                }
+                return GetBlock(Height);
```
Comment (shargon) [inline neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.cs]: Maybe in the `else` was good option set `persisting_block=GetBlock (Height)`

**Comment by erikzhang** [comment]
I'm not going to accept `Neo.Runtime.GetCurrentBlock` because we're going to implement a pre-execution mechanism for applications.
In a pre-execution, the `hash` field and `next_consensus` field of the `persisting_block` are not determined. If we want to ensure that the result of the pre-execution is consistent with the result of normal execution, we can't provide an API for getting `persisting_block`.

**Comment by localhuman** [comment]
Will remove `GetCurrentBlock`.  
for `GetTime`, the issue says to get the current block and add 15s.  Do we want to add that, given that the time may not be exact?

**Comment by erikzhang** [comment]
For `GetTime`:

1. If it's running under `Application` trigger, return the timestamp of the `persisting_block`.
1. If it's running under `Verification` trigger, return the timestamp of latest block and add `Blockchain.SecondsPerBlock`.

File: neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.cs (hunk 1 above)
Comment (localhuman) [reply neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.cs]: I want `persisting_block` to be `null` unless execution is in the middle of `LeveldbBlockchain.Persist`

**Pull request closed without merging**